agent/config/file_watcher.go

```diff
@@ -179,6 +204,8 @@ func (w *FileWatcher) handleEvent(ctx context.Context, event fsnotify.Event) err
 
 func (w *FileWatcher) isWatched(filename string) (*watchedFile, string, bool) {
 	path := filename
+	w.configFilesLock.RLock()
+	defer w.configFilesLock.RUnlock()
 	configFile, ok := w.configFiles[path]
```

Comment:
```suggestion
	w.configFilesLock.RLock()
	configFile, ok := w.configFiles[path]
	w.configFilesLock.RUnlock()
```

Avoid having a lingering `defer unlock` for longer than necessary so you don't deadlock with line 222.